### PR TITLE
init fuzzing infrastructure

### PR DESCRIPTION
**Pull request by franziskuskiefer**
This PR adds the basic infrastructure for fuzzing in #154.
I'm skipping the GH action for spot fuzzing for now, the setup is a little weird. I'll add that later.

The key package fuzzer will quickly fail. This will need more work as sketched in the changes to ciphersuite already.

### CODE REVIEW

File: src/ciphersuite/ciphersuites.rs [hunk 1]
```diff
@@ -14,24 +14,27 @@
...
-            0x0005 => CiphersuiteName::MLS10_256_DHKEMP521_AES256GCM_SHA512_P521,
-            0x0006 => CiphersuiteName::MLS10_256_DHKEMX448_CHACHA20POLY1305_SHA512_Ed448,
-            _ => panic!("Not implemented."),
+            0x0001 => Ok(CiphersuiteName::MLS10_128_DHKEMX25519_AES128GCM_SHA256_Ed25519),
+            0x0002 => Ok(CiphersuiteName::MLS10_128_DHKEMP256_AES128GCM_SHA256_P256),
+            0x0003 => Ok(CiphersuiteName::MLS10_128_DHKEMX25519_CHACHA20POLY1305_SHA256_Ed25519),
+            0x0004 => Ok(CiphersuiteName::MLS10_256_DHKEMX448_AES256GCM_SHA512_Ed448),
+            0x0005 => Ok(CiphersuiteName::MLS10_256_DHKEMP521_AES256GCM_SHA512_P521),
+            0x0006 => Ok(CiphersuiteName::MLS10_256_DHKEMX448_CHACHA20POLY1305_SHA512_Ed448),
+            _ => Err(Error::DecodingError),
```
Comment (kkohbrok) [inline src/ciphersuite/ciphersuites.rs:37]: Do we explicitly want errors when fuzzing as opposed to panics? Or What is the motivation for this change?

**Review by kkohbrok — APPROVED** [review]

File: src/ciphersuite/ciphersuites.rs (hunk 1 above)
Comment (franziskuskiefer) [reply src/ciphersuite/ciphersuites.rs:37]: We generally don't want to panic. Fuzzing is running into all the error cases. Panicking will be an error.
